R/utilities.R

```diff
@@ -127,19 +127,113 @@ get_ssh_session = function(host="gphost01.bcgsc.ca"){
 #'
 gene_to_region = function(gene_symbol,
                           ensembl_id,
-                          genome_build){
+                          genome_build = "grch37",
+                          return_as = "bed"){
 
-  if(genome_build != "grch37"){
-    message("Currently, only grch37 is supported")
-    return()
+  if(genome_build == "grch37"){
+    if(!missing(gene_symbol) && missing(ensembl_id)){
+      gene_coordinates = dplyr::filter(grch37_gene_coordinates, hugo_symbol %in% gene_symbol)
+    }
+
+    if(missing(gene_symbol) && !missing(ensembl_id)){
+      gene_coordinates = dplyr::filter(grch37_gene_coordinates, ensembl_gene_id %in% gene_symbol)
+    }
+  }
+
+  if(genome_build == "hg38"){
+    if(!missing(gene_symbol) && missing(ensembl_id)){
+      gene_coordinates = dplyr::filter(hg38_gene_coordinates, gene_name %in% gene_symbol)
+    }
+
+    if(missing(gene_symbol) && !missing(ensembl_id)){
+      gene_coordinates = dplyr::filter(hg38_gene_coordinates, ensembl_gene_id %in% gene_symbol)
+    }
+  }
+
+  if(return_as == "bed"){
+    region = paste0(gene_coordinates$chromosome, ":", gene_coordinates$start, "-", gene_coordinates$end)
+  }else if(return_as == "df"){
+    region = dplyr::select(gene_coordinates, chromosome, start, end, gene_name, hugo_symbol, ensembl_gene_id) %>%
+      as.data.frame() %>%
+      dplyr::arrange(chromosome, start)
   }
+
   if(!missing(gene_symbol)){
-    gene_coordinates = dplyr::filter(grch37_gene_coordinates, hugo_symbol == gene_symbol)
+    message(paste0(nrow(region), " region(s) returned for ", length(gene_symbol), " gene(s)"))
+  }
+
+  if(!missing(ensembl_id)){
+    message(paste0(nrow(region), " region(s) returned for ", length(ensembl_id), " gene(s)"))
   }
-  region = paste0(gene_coordinates$chromosome, ":", gene_coordinates$start, "-", gene_coordinates$end)
+
   return(region)
 }
 
+#' Return gennes residing in defined region(s)
+#'
+#' @param region Regions to intersect genes with, this should be a bed-like df (chromosome, start, end).
```

Comment:
We should let the user specify either a vector of regions (like the default output of gene_to_region) or this data frame for flexibility. There are some other functions that work the same way (get_ssm_by_region, maybe?)

Reply:
Yes, I am working on some updates to this function based on the comments regarding gene_to_region (since they are really similar, this one is basically the reverse version of the other function)